queue/abstract.lua

```diff
@@ -125,6 +125,16 @@ function tube.take(self, timeout)
     end
 end
 
+--- Checks is the task has been taken in a session with
+-- connection id == "conn_id".
+-- Throw an error on failure.
+local function check_task_on_taken(tube_id, task_id, conn_id)
+    local _taken = box.space._queue_taken:get{conn_id, tube_id, task_id}
+    if _taken == nil then
+        error("Task was not taken in the session")
+    end
+end
```

Comment:
Several nits, feel free to ignore.

1. I would not mix helpers with API function. I think it is better to move helpers at top.
2. Typo: 'Checks is' → 'if'.
3. 'Throw an error on failure' — it is unclear from the description, whether taken or not taken task is considered as failure.
4. `check_task_on_taken()` sounds strange. Variants: `check_task_is_taken()`, `check_task_is_owned()`, `check_task_ownership()`. Not sure here.
5. Since there is a usage of the function as a predicate, I would split it to `task_is_taken()` and `error_task_is_not_taken()` functions or kinda. Pcalling of a simple helper looks dubios for me. Alternative: `task_is_taken()` / `task_is_owned()`, which returns `is_taken, err`, so it may be placed directly to `assert()`.

Reply:
> Several nits, feel free to ignore.
> 
> 1. I would not mix helpers with API function. I think it is better to move helpers at top.

Accepted.

> 2. Typo: 'Checks is' → 'if'.

"If the task has been taken in a session...."? Are you sure?

> 3. 'Throw an error on failure' — it is unclear from the description, whether taken or not taken task is considered as failure.

Accepted.

> 4. `check_task_on_taken()` sounds strange. Variants: `check_task_is_taken()`, `check_task_is_owned()`, `check_task_ownership()`. Not sure here.

Yep. But `is` typically returns bool.

> 5. Since there is a usage of the function as a predicate, I would split it to `task_is_taken()` and `error_task_is_not_taken()` functions or kinda. Pcalling of a simple helper looks dubios for me. Alternative: `task_is_taken()` / `task_is_owned()`, which returns `is_taken, err`, so it may be placed directly to `assert()`.

I just moved the common logic into a separate function. If we add a wrapper, it changes the current behavior.